### ISSUE TITLE
Local functions don't report warnings for NotNull/NotNullWhen/NotNullIfNotNull

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#47896]
**Version Used**: d148f06

**Steps to Reproduce**:

```cs
#nullable enable
using System.Diagnostics.CodeAnalysis;

public class C
{
    public void M0(string? x)
    {
        local(ref x);
        
        void local([NotNull] ref string? x)
        {
            x = null;
        } // no warning, but one was expected
    }
    
    void M1([NotNull] ref string? x)
    {
        x = null;
    } // warning CS8777: Parameter 'x' must have a non-null value when exiting.
}
```
**Expected Behavior**: Warnings on both member methods and local functions with `[NotNull]` parameters

**Actual Behavior**: No warning on local function

We need to call functions like enforceNotNull, enforceMemberNotNull etc. here: https://github.com/dotnet/roslyn/blob/7e89bce35497ac106ed58621a281598bd2d6b2af/src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs#L2525

**Comment by jcouv** [comment]
Resolved in 17.0p5
